stream-chat-android-offline/src/main/java/io/getstream/chat/android/livedata/extensions/ClientExtensions.kt

```diff
@@ -137,6 +141,24 @@ public fun ChatError.isPermanent(): Boolean {
     return isPermanent
 }
 
+public fun ChatClient.domain(): ChatDomain {
+
+    // The builder/config logic happens at the ChatClient level
+    val domainBuilder = ChatDomain.Builder(appContext, this).setConfig(offlineConfig)
+
+    // We use listeners to tie into the ChatClient's setUser and disconnect flow
+    //
+    // // hookup listeners for pre set User and disconnect()
+    // setPreUserConnectListener {
```

Comment:
We could write it more polymorphic. Provide users the possibility to enhance our SDKs by their own extensions.
We could create such an interface:
``` kotlin
interface ChatExtensionsModule /* whatever name */ {
  /* add here some lifecycle methods */
fun onSetUserPendind(user: User)
fun onSetUserCompleted(user: User)
fun onSetUserFailed(user: User, exception: Throwable)
fun onConnected()
fun onDisconnected()
...
}
```

We could even split such interface to several (i.e. `ChatConnectionListener`, `UserAuthorizationListener` etc) and make ChatExtensionsModule as a Facade.
Then add to ChatClient method `addExtension(moduleExtension: ChatModuleExtension)`. Then `ChatClient` will have a list of such extensions that it will notify about an event of the lifecycle.